main/main.go

```diff
@@ -71,12 +75,16 @@ func runCmd() error {
 		}
 	}
 
-	// Stop early, is the dry run (-dry-run) flag was given.
-	if *flagDryRun {
+	// Stop early if the dry run (-dry-run) flag was given without an output-dir.
```

Comment:
Similar to the above remark, I think the code would become clearer if you avoided overloading flags, or relying on implications of flag values rather than their explicit meaning. Verify that flag settings are consistent once (i.e., either dry run or GCS bucket is specified, and the way I see it, output dir also only makes sense in conjunction with dry run), but then only rely on the individual flag. Concretely:
- I don't see a reason to exit early here, as the remainder is a no-op anyway
- The log message in l.85 should be based on `if *flagOutputDir != ""`, the same goes for the write instruction in l.100.